### PR TITLE
JITM: move jetpack-jitm.js to the JITM package

### PR DESCRIPTION
**Pull request by kbrown9**
#### Changes proposed in this Pull Request:

The JITM package should be able to be used independently of Jetpack. So, let's move the jetpack-jitm.js file out of Jetpack and into the JITM package.

* Update the monorepo:
    * Build the JITM package in the package.json `build_packages` and `build_production_packages` scripts.
    * Add `jetpack-jitm.js` to the eslint ignore list.
* Update the JITM package files:
    * Move the `jetpack-jitm.js` file to the JITM package.
    * Add a `webpack.config.packages.js` file.
    * Update the `.gitattributes` file.
    * Update the `.gitignore` file.
    * Add the `build-production` and `build-development` scripts to the composer.json file.
    * Add a `package.json` file.
    * Update the location of the jetpack-jitm.js file in `JITM::jitm_enqueue_files`.
    

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install and activate this branch and connect to wpcom.
2. Navigate to wp-admin -> Jetpack -> Dashboard. If you haven't already dismissed it, the `Give your site a speed boost` JITM should be displayed. Verify that it looks correct.

#### Proposed changelog entry for your changes:
* n/a - I don't think a changelog entry is needed for these changes.

### CODE REVIEW

File: projects/packages/jitm/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		"install-if-deps-outdated": "yarn install --check-files --production=false --frozen-lockfile",
+		"validate-es5": "eslint --parser-options=ecmaVersion:5 --no-eslintrc --no-ignore"
+	},
+	"devDependencies": {
+		"eslint": "7.18.0",
+		"webpack": "4.46.0",
+		"webpack-cli": "4.5.0"
+	},
+	"engines": {
+		"node": "^12.18.0",
```
Comment (jeherve) [inline projects/packages/jitm/package.json:30]: Can we bump this to `12.20.1` now that the monorepo requires this version of node (see `.nvmrc`)?

File: projects/packages/jitm/src/class-jitm.php [hunk 2]
```diff
@@ -115,7 +115,7 @@ public function jitm_enqueue_files() {
 
 		wp_enqueue_script(
 			'jetpack-jitm-new',
-			Assets::get_file_url_for_environment( '_inc/build/jetpack-jitm.min.js', '_inc/jetpack-jitm.js' ),
+			Assets::get_file_url_for_environment( 'assets/js/jetpack-jitm.min.js', 'assets/js/jetpack-jitm.js', __DIR__ ),
```
Comment (jeherve) [inline projects/packages/jitm/src/class-jitm.php:118]: In the Lazy Images package, we opted to have the assets inside `src/`, and here we have them in their own `assets/` dir. Would it make sense to move everything inside of `src/` now?

File: projects/packages/jitm/package.json (hunk 1 above)
Comment (kbrown9) [reply projects/packages/jitm/package.json]: Fixed in commit 803c759.

File: projects/packages/jitm/src/class-jitm.php (hunk 2 above)
Comment (kbrown9) [reply projects/packages/jitm/src/class-jitm.php]: In commit 803c759, I moved the css files to `src/css` and the js file to `src/js`.

**Review by jeherve — APPROVED** [review]
This looks to me now, and tests well! 🚢

**Review by jeherve — CHANGES_REQUESTED** [review]
Coming back on my review, sorry. I think we'll need to update https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/tools/builder/sass.js#L259 now that the CSS location has changed.

**Comment by kbrown9** [comment]
> Coming back on my review, sorry. I think we'll need to update
> 
> https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/tools/builder/sass.js#L259
> 
> now that the CSS location has changed.

Good catch! I updated that file in commit f6843ea. I also updated the [file location](https://github.com/Automattic/jetpack/pull/18341/commits/f6843ea8adf1e400104b25c5436c6b970c6f3dc1#diff-cc3950066f06f29e4d2bb142dce3f1c059262a074bbe25757ab33e27cdfd20d9L61) in `projects/plugins/jetpack/.gitattributes`. Please take a look at those changes when you get a chance!

**Review by jeherve — APPROVED** [review]
This looks good to me. 🚢

**Comment by jeherve** [comment]
r226390-wpcom

**Pull request merged**